app/src/main/java/app/musikus/usecase/activesession/GetSessionTimerState.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright (c) 2024 Michael Prommersberger
+ *
+ */
+
+package app.musikus.usecase.activesession
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+enum class SessionTimerState {
```

Comment:
Do we really need a separate `SessionTimerState`? Can't we consolidate it in a single `ActiceSessionState`?

Reply:
Yes, this bothers me also a bit. I thought of the state as a kind of *domain-level* state which gives us a `Flow` to observe events like pausing, starting, etc. to consume it in the UI. The `sessionState` in `ActiveSessionViewModel` is then a *UI-state*, but [currently there is a 1:1 mapping](https://github.com/matthiasemde/musikus-android/blob/1d6f8c1c873919c9158ec292a3d9ddbaf811c1c3/app/src/main/java/app/musikus/ui/activesession/ActiveSessionViewModel.kt#L75), so kind of only a proxy.

Maybe in the future we want to add functionality which is only ui-specific and not domain-specific or so. If we now ditch the *ui state* object, only the `when` (and the enum class) would not be needed anymore, so a total gain of 10 LOC we're talking about 😅

Reply:
As I said in my other comments I think we should differentiate at least three kinds of state: 
- _domain/data-state_ which is more or less persistent.
- _ViewModel-state_ which are StateFlows stored in and managed by the ViewModel
- _UI-state_ which is entirely derived by the previous two types of state 

Now it might be legit to have a separate `ActiveSessionState` on _domain_ and _ViewModel_ level but I'm not sure whether we need it at the moment.

Reply:
I don't see what is the difference between ViewModel and UI state. The ViewModel should contain the UI state

Reply:
Yes, and no. In our case the ViewModel contains the logic which is used to derive the UiState from the data coming from the domain layer + the non persistent state which the ViewModel stores itself.

I concede that saying the UiState is entirely derived is kind of paradoxical since then it isn't really state anymore but still...